### ISSUE TITLE
SequentialLR with a Tensor view produces an unexpected learning-rate value

### ISSUE THREAD

**Issue by Wanzy0209** [pytorch/pytorch#195984]
## Problem description

SequentialLR is initialized with a Tensor learning rate that is a view of another Tensor. The test expects the visible base Tensor, initial_lr, and chained scheduler base_lrs to remain consistent after scheduler initialization and stepping.

## Test case

```python
import torch
from torch import optim

def test_sequential_lr_with_tensor_view():

    base_lr = torch.tensor([1.0])
    lr_view = base_lr.view(-1)

    x = torch.tensor(0.0, device='meta')
    opt = optim.AdamW([x], lr=lr_view)

    milestone, total_steps = 40, 100
    start_factor, end_factor = 0.2, 1.0
    warmup = optim.lr_scheduler.LinearLR(opt, start_factor, end_factor, total_iters=milestone)
    decay = optim.lr_scheduler.CosineAnnealingLR(opt, T_max=total_steps-milestone)

    scheduler = optim.lr_scheduler.SequentialLR(opt, schedulers=[warmup, decay], milestones=[milestone])

    scheduler.step()

    assert base_lr.item() == start_factor, \
        f"Base LR (via view) should be updated to {start_factor}, got {base_lr.item()}"

    assert opt.param_groups[0]['initial_lr'].item() == 1.0, \
        f"initial_lr should not be corrupted, expected 1.0, got {opt.param_groups[0]['initial_lr'].item()}"

    assert decay.base_lrs[0].item() == 1.0, \
        f"Chained scheduler base_lrs should not be corrupted, expected 1.0, got {decay.base_lrs[0].item()}"

    assert opt.param_groups[0]['lr'].data_ptr() != opt.param_groups[0]['initial_lr'].data_ptr(), \
        "lr and initial_lr should not be aliased to prevent corruption."

    print("Test passed: SequentialLR correctly handles Tensor learning rates without corrupting initial_lr.")

if __name__ == "__main__":
    test_sequential_lr_with_tensor_view()
```

## Error information

**Failed with a different value**. The historical run observed 1.0 where 0.2 was expected. PyTorch 2.13.0+cpu observed 0.2200000137 after the explicit scheduler.step() call.

### Versions

PyTorch version: 2.13.0+cpu
Is debug build: False
CUDA used to build PyTorch: None
ROCM used to build PyTorch: N/A

OS: Windows 10 Pro 25H2 (10.0.26200.8655 AMD64)
GCC version: Could not collect
Clang version: Could not collect
CMake version: Could not collect
Libc version: N/A

Python version: 3.11.6 (tags/v3.11.6:8b6ee5b, Oct  2 2023, 14:57:12) [MSC v.1935 64 bit (AMD64)] (64-bit runtime)
Python platform: Windows-10-10.0.26200-SP0
Is CUDA available: False
CUDA runtime version: No CUDA
CUDA_MODULE_LOADING set to: N/A
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
Is XPU available: False
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: False
Caching allocator config: N/A

CPU:
Intel64 Family 6 Model 151 Stepping 2, GenuineIntel

Versions of relevant libraries:
[pip3] numpy==2.4.6
[pip3] optree==0.20.0
[pip3] torch==2.13.0
[conda] Could not collect

**Comment by abhinav-phi** [comment]
Hi! I'd like to work on this.

Plan: I'll investigate the reported behavior, implement the change/fix per the issue description, and add tests to cover it.

Could you please assign this issue to me? Thanks!

/assign take
